src/main/java/com/example/hashcartapp/service/AdvertisementService.java

```diff
@@ -0,0 +1,121 @@
+package com.example.hashcartapp.service;
+
+import com.example.hashcartapp.dto.AdvertisementDTO;
+import com.example.hashcartapp.entities.Advertisement;
+import com.example.hashcartapp.repository.AdvertisementRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static org.hibernate.internal.CoreLogging.logger;
+
+@Service
+public class AdvertisementService {
+
```

Comment:
Please define a service interface, and then create a service implementation
controller -> service interface -> service impl -> repository